vault/diagnose/tls_verification_test.go

```diff
@@ -0,0 +1,317 @@
+package diagnose
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/vault/internalshared/configutil"
+	"github.com/hashicorp/vault/internalshared/listenerutil"
+)
+
+// TestTLSValidCert is the positive test case to show that specifying a valid cert and key
+// passes all checks.
+func TestTLSValidCert(t *testing.T) {
+	listeners := []listenerutil.Listener{
+		{
+			Config: &configutil.Listener{
+				Type:                          "tcp",
+				Address:                       "127.0.0.1:443",
+				ClusterAddress:                "127.0.0.1:8201",
+				TLSCertFile:                   "./test-fixtures/goodcertwithroot.pem",
+				TLSKeyFile:                    "./test-fixtures/goodkey.pem",
+				TLSMinVersion:                 "tls10",
+				TLSRequireAndVerifyClientCert: true,
+				TLSDisableClientCerts:         false,
+			},
+		},
+	}
+	err := ListenerChecks(listeners)
+	if err != nil {
+		t.Error(err.Error())
+	}
+}
+
+// TestTLSFakeCert simply ensures that the certificate file must contain PEM data.
+func TestTLSFakeCert(t *testing.T) {
+	listeners := []listenerutil.Listener{
+		{
+			Config: &configutil.Listener{
+				Type:                          "tcp",
+				Address:                       "127.0.0.1:443",
+				ClusterAddress:                "127.0.0.1:8201",
+				TLSCertFile:                   "./test-fixtures/fakecert.pem",
+				TLSKeyFile:                    "./../../api/test-fixtures/keys/key.pem",
+				TLSMinVersion:                 "tls10",
+				TLSRequireAndVerifyClientCert: true,
+				TLSDisableClientCerts:         false,
+			},
+		},
+	}
+	err := ListenerChecks(listeners)
+	if err == nil {
+		t.Error("TLS Config check on fake certificate should fail")
```

Comment:
i think these should be fails (not errors), since you can't do the next check if `err` is nil (presumably that's a caught panic with the test infra)

Reply:
there are multiple cases like this

Reply:
I'm so sorry! I didn't realized you had left some comments @swayne275 with the approval. Yeah, fails here would be better. I'll push a PR out today cleaning that up. 

I can also use `:=` for the error var in the cleanup PR. For the `!string.Contains()` comment below, I can't do that for this particular test, because the error message thrown by the library is a concatenation of a few error messages. I didn't want to copy that whole string here, so I just checked the necessary part to distinguish this failure from the other cases.

Thanks for the comments, and sorry about the premature merge! I'll have followup PR with the improvements ASAP!

Reply:
No problem! I wouldn't have approved it if I wasn't okay with it going without fixes/responses to the comments

The panic in tests isn't that big of a deal, it's just a lot cleaner to avoid it

Reply:
I'm changing these to Fatals, not Fails -- Error is equivalent to a Log followed by a Fail, and Fail continues execution. Fatal stops the test with a FailNow.

Reply:
oh yeah fatal is probably what i meant, cool!